### PR TITLE
🐛 Fixes issue with browser back not closing lightbox

### PR DESCRIPTION
**Pull request by joaopsrleal**
Fixes issue with lightbox component not properly handling the browser back button to close itself, reported in #30136. 

This was being caused because we were pushing the close(), which expects an ActionTrust as an argument, call directly to the history stack, even though the [doPop method](https://github.com/ampproject/amphtml/blob/f3712ad79e4aba79a0b4ec6b5edd79d3c9bf868b/src/service/history-impl.js#L236) passes a historyState object.

### CODE REVIEW

File: extensions/amp-lightbox/0.1/amp-lightbox.js
Comment (kristoferbaxter) [inline extensions/amp-lightbox/0.1/amp-lightbox.js:411]: My hunch is this should not override the trust provided to the method.

`.push((unused) => this.close(trust))`

Adding @caroqliu for a second opinion.

Comment (joaopsrleal) [reply extensions/amp-lightbox/0.1/amp-lightbox.js:411]: That's a good point. I was on the fence here, since this will be executed as a result of an explicit user action (clicking browser back). But I can see how it would also make sense to respect the provided trust.

I'll wait for @caroqliu's opinion before changing it.

Comment (caroqliu) [reply extensions/amp-lightbox/0.1/amp-lightbox.js:411]: The method definition for `close` doesn't appear to gate based on a certain minimum trust, so it should not make a difference what the value is. Though I agree there is sufficient user intention to imply `ActionTrust.HIGH`, it may be more confusing to override the given trust without any change in end user experience.

Comment (joaopsrleal) [reply extensions/amp-lightbox/0.1/amp-lightbox.js]: That sounds good. I've updated the code to pass in the provided trust value.

**Review by caroqliu — APPROVED** [review]

**Comment by joaopsrleal** [comment]
@kristoferbaxter the PR seems to have failing visual tests which seem to be unrelated. Could this be flakiness in the test? Any ideas on how to overcome the failing check?

**Comment by joaopsrleal** [comment]
> @kristoferbaxter the PR seems to have failing visual tests which seem to be unrelated. Could this be flakiness in the test? Any ideas on how to overcome the failing check?

Judging by https://percy.io/ampproject/amphtml/builds/7428298, it seems like these tests are indeed flaky.

In the first case, I imagine that sometimes 1s (see [wait config](https://github.com/ampproject/amphtml/blob/e8722bbafa48b5ae218d0f4ecc46a4826100cb77/test/fixtures/e2e/amphtml-ads/amp-creative.html#L52) here) is not enough for the image of the last ad to render. Or at least it wasn't for the golden snapshot.

In the second case, the amp-ad element on the top of the page has a running animation. So the test will always be flaky because a small difference in the timing of the screenshot will cause the spinner to be in a different position. So we'd need to somehow either change the test case to have a static loading icon.

Any ideas of who would be the best folks to help out with these tests?

**Comment by Beethoven** [comment]
Any news on Merging?

**Comment by caroqliu** [comment]
@joaopsrleal Thanks for your patience and communication here! I have manually approved the visual diff tests, and the code coverage test is non-blocking. Is this PR okay to merge?

**Comment by joaopsrleal** [comment]
> @joaopsrleal Thanks for your patience and communication here! I have manually approved the visual diff tests, and the code coverage test is non-blocking. Is this PR okay to merge?

No worries! Thank you for taking a look at the failures.
From my side, it should be good to go. Hopefully there will be no merge conflicts, but I'm happy to merge with the latest master state if needed.

**Pull request merged**
